packages/react/src/next/KnockGuideLocationSensor.tsx

```diff
@@ -0,0 +1,71 @@
+"use client";
+
+import { useGuideContext } from "@knocklabs/react-core";
+// Note the .js suffix for Next 15 ESM friendliness:
+import { usePathname, useSearchParams } from "next/navigation.js";
+import { useRouter } from "next/router.js";
+import React from "react";
+
+import { checkForWindow } from "../modules/core/utils";
+
+const PagesRouter: React.FC = () => {
+  const router = useRouter();
+  const { client } = useGuideContext();
+
+  React.useEffect(() => {
+    const win = checkForWindow();
+    if (!win) return;
+
+    // Set the initial location if not yet set.
+    if (!client.store.state.location) {
+      client.setLocation(win.location.href);
+    }
+
+    // Remove any location chagne event listeners on the window object in case
+    // they are attached.
+    client.removeLocationChangeEventListeners();
```

Comment:
the point of this is to remove stale event listeners? maybe a "why we need to do this" comment would be helpful

Reply:
So [by default](https://github.com/knocklabs/javascript/blob/main/packages/client/src/clients/guide/client.ts#L283) `KnockGuideProvider` sets `trackLocationFromWindow` to `true` which attaches event listeners for "popstate" and "hashchange" on the global history object. 

If you are using this `KnockGuideLocationSensor` helper module, that means we are detecting route changes from Next's router rather than from the global history object, so we just want to avoid any event listeners hanging around unnecessarily.

Reply:
cool! I think it's worthwhile to paste that right in a code comment there